### PR TITLE
Adding customerAdminGroupId to OSA cluster swagger spec

### PR DESCRIPTION
**Pull request by JackQuincy**
### Latest improvements:
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i><br>
### Contribution checklist:
- [x] I have reviewed the [documentation](https://github.com/Azure/azure-rest-api-specs#basics) for the workflow.
- [x] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.
- [x] The [OpenAPI Hub](https://aka.ms/openapiportal) was used for checking validation status and next steps.
### ARM API Review Checklist
- [ ] Service team MUST add the "WaitForARMFeedback" label if the management plane API changes fall into one of the below categories. 
- adding/removing APIs.
- adding/removing properties.
- adding/removing API-version. 
- adding a new service in Azure.

Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs.
- [ ] If you are blocked on ARM review and want to get the PR merged urgently, please get the ARM oncall for reviews (RP Manifest Approvers team under Azure Resource Manager service) from IcM and reach out to them. 
Please follow the link to find more details on [API review process](https://armwiki.azurewebsites.net/rp_onboarding/ResourceProviderOnboardingAPIRevieworkflow.html).

### CODE REVIEW

**Comment by JackQuincy** [comment]
This should have WaitForARMFeedback label but I can't add it

**Comment by adxsdknet** [comment]
# Automation for azure-sdk-for-net 
A PR has been created for you: 
https://github.com/azure/azure-sdk-for-net/pull/5611 
.NET SDK Commits: 
https://github.com/adxsdknet/azure-sdk-for-net/pull/5611/commits/7709dd7afd31491038a8b01ab500f78f24bf8d36

File: specification/containerservice/resource-manager/Microsoft.ContainerService/preview/2018-09-30-preview/examples/OpenShiftManagedClustersGet.json [hunk 1]
```diff
@@ -60,7 +60,8 @@
                 "provider": {
                   "kind": "AADIdentityProvider",
                   "clientId": "clientId",
-                  "tenantId": "tenantId"
+                  "tenantId": "tenantId",
+                  "customerAdminGroupId": "customerAdminGroupId"
```
Comment (majastrz) [inline specification/containerservice/resource-manager/Microsoft.ContainerService/preview/2018-09-30-preview/examples/OpenShiftManagedClustersGet.json]: >customerAdminGroupId [](start = 43, length = 20)

This is kind of misleading since the value is a GUID. The way this is typically represented in the Swagger is by either enclosing the content in curly braces ("{customerAdminGroupId}", for example) or by just listing empty GUID in the format your service returns ("00000000-0000-0000-0000-000000000000", for example). I have the same comment for all the AAD-related properties in all the existing examples.

File: specification/containerservice/resource-manager/Microsoft.ContainerService/preview/2018-09-30-preview/openShiftManagedClusters.json [hunk 2]
```diff
@@ -731,6 +731,10 @@
             "tenantId": {
               "type": "string",
               "description": "The tenantId associated with the provider."
+            },
+            "customerAdminGroupId": {
```
Comment (majastrz) [inline specification/containerservice/resource-manager/Microsoft.ContainerService/preview/2018-09-30-preview/openShiftManagedClusters.json]: >customerAdminGroupId [](start = 13, length = 20)

How will this impact existing clients. What will the behavior be when it's missing? Any back-compat concerns?

**Review by majastrz — CHANGES_REQUESTED** [review]
Can you update the examples and answer my question about back-compat?

File: specification/containerservice/resource-manager/Microsoft.ContainerService/preview/2018-09-30-preview/examples/OpenShiftManagedClustersGet.json (hunk 1 above)
Comment (JackQuincy) [reply specification/containerservice/resource-manager/Microsoft.ContainerService/preview/2018-09-30-preview/examples/OpenShiftManagedClustersGet.json]: updated

File: specification/containerservice/resource-manager/Microsoft.ContainerService/preview/2018-09-30-preview/openShiftManagedClusters.json (hunk 2 above)
Comment (JackQuincy) [reply specification/containerservice/resource-manager/Microsoft.ContainerService/preview/2018-09-30-preview/openShiftManagedClusters.json]: Creates with the field missing will just not use/get the new feature.  Clients should be able to handle receiving an unexpected field and just ignore the new field on gets. If they do an update call and the field is missing we will just not update the field and it won't undo work done with a newer client.

Comment (JackQuincy) [reply specification/containerservice/resource-manager/Microsoft.ContainerService/preview/2018-09-30-preview/openShiftManagedClusters.json]: In simpler terms. This is an optional field. If it is missing old behavior is unchanged. If we add the field it will get the new functionality. With this being a preview version this should be fine. CC: @amanohar

Comment (majastrz) [reply specification/containerservice/resource-manager/Microsoft.ContainerService/preview/2018-09-30-preview/openShiftManagedClusters.json]: Makes sense.

---
In reply to: [270517299](https://github.com/Azure/azure-rest-api-specs/pull/5491#discussion_r270517299) [](ancestors = 270517299)

**Review by majastrz — APPROVED** [review]
Looks good.

**Comment by JackQuincy** [comment]
@praries880 or @majastrz  can this get merged?

**Comment by majastrz** [comment]
> @praries880 or @majastrz can this get merged?

@praries880 can merge it in. The ARM side doesn't have permissions to merge this.
